src/anyio/_backends/_asyncio.py

```diff
@@ -2679,17 +2683,18 @@ async def wait_socket_readable(cls, sock: socket.socket) -> None:
             read_events = {}
             _read_events.set(read_events)
 
-        if read_events.get(sock):
+        sock_fileno = sock.fileno()
```

Comment:
We should be able to remove the wait_socket_writable and wait_socket_readable from the backend as they're not public

Reply:
You mean using `wait_readable` and `wait_writable` instead?

Reply:
Yep